spring-cloud-dataflow-composed-task-runner/src/main/java/org/springframework/cloud/dataflow/composedtaskrunner/TaskLauncherTasklet.java

```diff
@@ -214,6 +216,17 @@ else if (taskExecution.getExitCode() != 0) {
 		return RepeatStatus.CONTINUABLE;
 	}
 
+	public String getParentTaskExecutionId(StepContribution stepContribution) {
+		Long result = null;
+		if (this.taskProperties.getExecutionid() != null) {
+			result = this.taskProperties.getExecutionid();
+		}
+		else if (stepContribution != null) {
+			result = this.taskExplorer.getTaskExecutionIdByJobExecutionId(stepContribution.getStepExecution().getJobExecutionId());
```

Comment:
Just to play devil's advocate... in the case where `getJobExecutionId` returns `null`, the underlying DAO will issue:
```sql
SELECT TASK_EXECUTION_ID FROM %PREFIX%TASK_BATCH WHERE JOB_EXECUTION_ID = null
```
which seems like it could return other unwanted results.

Reply:
~which seems like it could return other unwanted results.~

Actually I think we are ok here because it is using `queryForObject` which will throw `IncorrectResultSizeDataAccessException` if there are more than 1 result. Hence if there is a `null` one and this succeeds it must be the `null` belonging to this task.